tests/org.eclipse.papyrus.uml.interaction.model.tests/src/org/eclipse/papyrus/uml/interaction/model/tests/creation/SemanticOrderAfterCreationOfElementOnTopTest.java

```diff
@@ -0,0 +1,196 @@
+package org.eclipse.papyrus.uml.interaction.model.tests.creation;
+
+import static org.eclipse.uml2.uml.UMLPackage.Literals.ACTION_EXECUTION_SPECIFICATION;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.emf.common.command.Command;
+import org.eclipse.papyrus.uml.interaction.model.CreationCommand;
+import org.eclipse.papyrus.uml.interaction.model.MExecution;
+import org.eclipse.papyrus.uml.interaction.model.MInteraction;
+import org.eclipse.papyrus.uml.interaction.model.MLifeline;
+import org.eclipse.papyrus.uml.interaction.model.MMessage;
+import org.eclipse.papyrus.uml.interaction.model.MMessageEnd;
+import org.eclipse.papyrus.uml.interaction.model.MOccurrence;
+import org.eclipse.papyrus.uml.interaction.model.tests.ModelEditFixture;
+import org.eclipse.papyrus.uml.interaction.tests.rules.ModelResource;
+import org.eclipse.uml2.uml.ExecutionSpecification;
+import org.eclipse.uml2.uml.Message;
+import org.eclipse.uml2.uml.MessageSort;
+import org.junit.Rule;
+import org.junit.Test;
+
+@SuppressWarnings("nls")
+@ModelResource({"SemanticOrderAfterCreationTest.uml", "SemanticOrderAfterCreationTest.notation" })
+public class SemanticOrderAfterCreationOfElementOnTopTest {
+
+	private static final String QN = "343-test::Interaction::%s";
+
+	@Rule
+	public final ModelEditFixture model = new ModelEditFixture();
+
+	private MInteraction interaction;
+
+	private MInteraction interaction() {
+		if (interaction == null) {
+			interaction = model.getMInteraction();
+		}
+		return interaction;
+	}
+
+	private void execute(Command command) {
+		model.execute(command);
+		/* force reinit after change */
+		interaction = null;
+	}
+
+	@Test
+	@SuppressWarnings({"boxing" })
```

Comment:
We should configure the test projects to ignore boxing.  The annotations are so much noise ...

Reply:
(y)